repose-aggregator/components/filters/openstack-identity-v3-filter/src/main/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3Handler.scala

```diff
@@ -87,6 +86,19 @@ class OpenStackIdentityV3Handler(identityConfig: OpenstackIdentityV3Config, iden
       response.addHeader(HttpHeaders.RETRY_AFTER, retry)
     }
 
+    implicit val catalogWrites = new Writes[ServiceForAuthenticationResponse] {
+      def writes(catalog: ServiceForAuthenticationResponse) = JsObject(Seq(
+        "endpoints" -> JsArray(catalog.endpoints.map(endpoint => JsObject(
+          Seq("id" -> JsString(endpoint.id)) ++
+            endpoint.name.map(name => Seq("name" -> JsString(name))).getOrElse(Seq()) ++
+            endpoint.interface.map(interface => Seq("interface" -> JsString(interface))).getOrElse(Seq()) ++
+            endpoint.region.map(region => Seq("region" -> JsString(region))).getOrElse(Seq()) ++
+            Seq("url" -> JsString(endpoint.url)) ++
+            endpoint.service_id.map(serviceId => Seq("service_id" -> JsString(serviceId))).getOrElse(Seq()))))) ++
+        catalog.id.map(id => Seq("id" -> JsString(id))).getOrElse(Seq()) ++
+        catalog.name.map(name => Seq("name" -> JsString(name))).getOrElse(Seq()))
+    }
```

Comment:
Yup, this is one of the things I plan on changing.

1. It's using the fancy implicit writes thing to be able to do `Json.toJson` later in the class.  I think I'm going to move the JSON construction down so it'll be closer to where it's being used and avoid using an implicit completely.  https://github.com/rackerlabs/repose/blob/70c346711aaa39ebd7fdaf6576b1a0ba45125b7e/repose-aggregator/components/filters/openstack-identity-v3-filter/src/main/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3Handler.scala#L235

2. It is using class construction.  I'm not sure what you mean???

3. If you're talking about the `Seq()`, the alternative doesn't look that great:
`Seq.empty[(String, JsValue)]`

4. When using JsObject(Seq(...)), there is no automatic conversion from String to JsString.  It must be specified.

4a. I'm using JsObject(Seq(...)) over Json.obj(...) since (in order to maintain the old behavior which may be something we don't need to do) I need to be able to leave out fields if they're going to be null.  It's still possible to do it using Json.obj(...), but it looks messier.

5. Yeah, I planned on fixing indentation issues like that.